Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -6768,7 +6768,11 @@ void HTMLMediaElement::enterFullscreen(VideoFullscreenMode mode)
     m_changingVideoFullscreenMode = true;
 
 #if ENABLE(FULLSCREEN_API) && ENABLE(VIDEO_USES_ELEMENT_FULLSCREEN)
-    if (document().settings().fullScreenEnabled() && mode == VideoFullscreenModeStandard) {
+    bool videoUsesElementFullscreen = true;
+#if PLATFORM(IOS_FAMILY)
+    videoUsesElementFullscreen &= document().settings().videoFullscreenRequiresElementFullscreen();
+#endif
```

Comment:
```suggestion
#if PLATFORM(IOS_FAMILY)
    bool videoUsesElementFullscreen = document().settings().videoFullscreenRequiresElementFullscreen();
#else
    bool videoUsesElementFullscreen = true;
#endif
```

 I have a slight preference for writing like this.

Reply:
Sure.

Reply:
Oh I actually had the same feedback 👍